eras/dijkstra/src/Cardano/Ledger/Dijkstra/TxBody.hs

```diff
@@ -465,22 +467,25 @@ instance EraTxBody DijkstraEra where
     getTotalRefundsTxCerts pp lookupStakingDeposit lookupDRepDeposit (txBody ^. certsTxBodyL)
 
 upgradeGovAction ::
-  Coercible (PParamsHKD StrictMaybe (PreviousEra era)) (PParamsHKD StrictMaybe era) =>
-  GovAction (PreviousEra era) -> GovAction era
-upgradeGovAction (ParameterChange x y z) = ParameterChange (coerce x) (coerce y) z
-upgradeGovAction (HardForkInitiation x y) = HardForkInitiation (coerce x) y
-upgradeGovAction (TreasuryWithdrawals x y) = TreasuryWithdrawals x y
-upgradeGovAction (NoConfidence x) = NoConfidence x
-upgradeGovAction (UpdateCommittee x y z w) = UpdateCommittee x y z w
-upgradeGovAction (NewConstitution x y) = NewConstitution x (coerce y)
-upgradeGovAction InfoAction = InfoAction
+  forall era.
+  (EraPParams era, EraPParams (PreviousEra era)) =>
+  UpgradePParams StrictMaybe era ->
+  GovAction (PreviousEra era) ->
+  GovAction era
+upgradeGovAction args (ParameterChange x y z) = ParameterChange (coerce x) (upgradePParamsUpdate args y) z
+upgradeGovAction _ (HardForkInitiation x y) = HardForkInitiation (coerce x) y
+upgradeGovAction _ (TreasuryWithdrawals x y) = TreasuryWithdrawals x y
+upgradeGovAction _ (NoConfidence x) = NoConfidence x
+upgradeGovAction _ (UpdateCommittee x y z w) = UpdateCommittee x y z w
+upgradeGovAction _ (NewConstitution x y) = NewConstitution x (coerce y)
+upgradeGovAction _ InfoAction = InfoAction
```

Comment:
I don't think it ever makes sense to use something other than empty pparams. In order to make it era agnostic we can use `emptyPParamsStrictMaybe` instead of `emptyUpgradeDijkstraPParams` 
```suggestion
upgradeGovAction (ParameterChange x y z) = ParameterChange (coerce x) (upgradePParamsUpdate emptyPParamsStrictMaybe y) z
upgradeGovAction (HardForkInitiation x y) = HardForkInitiation (coerce x) y
upgradeGovAction (TreasuryWithdrawals x y) = TreasuryWithdrawals x y
upgradeGovAction (NoConfidence x) = NoConfidence x
upgradeGovAction (UpdateCommittee x y z w) = UpdateCommittee x y z w
upgradeGovAction (NewConstitution x y) = NewConstitution x (coerce y)
upgradeGovAction InfoAction = InfoAction
```